http/logical.go

```diff
@@ -227,6 +245,11 @@ func buildLogicalRequestNoAuth(w http.ResponseWriter, r *http.Request) (*logical
 		Headers:    r.Header,
 	}
 
+	headerCert := handleCertHeader(r)
+	if headerCert != nil {
+		req.ClientHeaderCert = headerCert
```

Comment:
(Just cross-linking to discussion for others reviewing:  https://linuxfoundation.zulipchat.com/#narrow/channel/529890-openssf-openbao-discussion/topic/Forwarded.20Tls-Client-Cert/near/554761022)